runtime/src/accounts_db.rs

```diff
@@ -1410,12 +1409,18 @@ impl AccountsDB {
         let store =
             Arc::new(self.new_storage_entry(slot, &Path::new(&self.paths[path_index]), size));
         let store_for_index = store.clone();
-        let slot_storage = self
```

Comment:
DashMap returns RefMut here and the `.clone()` in new code makes it (and the write lock on DashMap backing it) drop as soon as it goes out of scope (well temporary expression in this case), right?

Could you comment about that?

Well, so hard finding and we need some commenting I guess.

Reply:
Anyway, thanks for finding these very naughty bugs!

Reply:
Added a comment and a type for clarity!